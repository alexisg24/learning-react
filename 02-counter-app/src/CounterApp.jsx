import { useState } from 'react'
import PropType from 'prop-types'

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value)
  const handleAdd = () => {
    setCounter(counter + 1)
  }

  const handleSubstract = () => {
    setCounter(counter - 1)
  }

  const handleReset = () => {
    setCounter(value)
  }

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <section>
        <button
          onClick={handleAdd}
        >
          +1
        </button>
        <button
          onClick={handleSubstract}
        >
          -1
        </button>
        <button
          aria-label='btn-reset'
          onClick={handleReset}
        >
          Reset
        </button>
      </section>

    </>
  )
}

CounterApp.propTypes = {
  value: PropType.number.isRequired
}

CounterApp.defaultProps = {
  value: 0
}

export { CounterApp }
