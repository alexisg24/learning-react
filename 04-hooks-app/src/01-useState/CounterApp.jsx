import { useState } from 'react'

const CounterApp = () => {
  const [counter, setCounter] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30
  })

  const { counter1, counter2, counter3 } = counter
  console.log(counter)
  return (
    <>
      <h1>Counter 1: {counter1} </h1>
      <h1>Counter 2: {counter2} </h1>
      <h1>Counter 3: {counter3} </h1>
      <hr />
      <button
        onClick={() => setCounter(prev => ({ ...prev, counter1: prev.counter1 + 1 }))}
        className='btn'
      >+1
      </button>
    </>
  )
}

export { CounterApp }
