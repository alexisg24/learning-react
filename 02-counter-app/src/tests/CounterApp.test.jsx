import { fireEvent, render, screen } from '@testing-library/react'
import { CounterApp } from '../CounterApp'

describe('Tests in <CounterApp />', () => {
  const initialValue = 100
  test('Should match with the snapshot', () => {
    const { container } = render(<CounterApp value={initialValue} />)
    expect(container).toMatchSnapshot()
  })

  test('Initial Value should be 100', () => {
    render(<CounterApp value={initialValue} />)
    expect(screen.getByText(initialValue)).toBeTruthy()
    expect(screen.getByRole('heading', { level: 2 }).innerHTML)
      .toContain(initialValue.toString())
  })

  test('+1 Button should increment inital value', () => {
    render(<CounterApp value={initialValue} />)
    fireEvent.click(screen.getByText('+1'))
    expect(screen.getByText('101')).toBeTruthy()
  })

  test('-1 Button should decrement value', () => {
    render(<CounterApp value={initialValue} />)
    fireEvent.click(screen.getByText('-1'))
    expect(screen.getByText('99')).toBeTruthy()
  })

  test('Reset Button should works', () => {
    render(<CounterApp value={initialValue} />)
    fireEvent.click(screen.getByText('+1'))
    fireEvent.click(screen.getByText('+1'))
    fireEvent.click(screen.getByText('+1'))
    // fireEvent.click(screen.getByText('Reset'))
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }))
    expect(screen.getByText(initialValue)).toBeTruthy()
  })
})
