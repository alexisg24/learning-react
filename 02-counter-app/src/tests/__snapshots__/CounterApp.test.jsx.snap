// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <CounterApp /> Should match with the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <section>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button
      aria-label="btn-reset"
    >
      Reset
    </button>
  </section>
</div>
`;
